comp/serializer/compression/def/component.go

```diff
@@ -33,3 +35,13 @@ type StreamCompressor interface {
 	io.WriteCloser
 	Flush() error
 }
+
+// Requires contains the config for Compression
+type Requires struct {
+	Cfg config.Component
+}
+
+// Provides contains the compression component
+type Provides struct {
+	Comp Component
+}
```

Comment:
These structs should be define in the `impl-*` packages. Any reason you defined here?

Reply:
The same ones are used in all the impls.  I can def copy it to each one but this seemed a little cleaner?

Reply:
To avoid having too many variations of how we defined and structure components I would advise to duplicate on each of the `impl-*` packages. 

The copy paste culture is strong here 😄 🍝

Reply:
sure lol

Reply:
Actually.. this is more difficult than that.  
So all the impls share a Provides.  Only one needs a Requires - the zstd one.  So far, I think that's pretty normal.

The reason I can't just copy paste them everywhere is that the `selector` module has its own Requires - one with the configuration.  Then, it *shares* the Provides with all the impls - that way each impl is providing the same type.  Any idea how to detangle this?  It's a little confusing...